hs/CalendarResponse.hs

```diff
@@ -0,0 +1,230 @@
+    thursday = let thur field = field !! 0 == 3.0  in filter thur lst
+    friday = let fri field = field !! 0 == 4.0 in filter fri lst
+
+-- ** Start time
+
+-- | Obtains a list of start times for each course.
+startTime :: [[[Double]]] -> [[String]]
+startTime weekFields = map checkTimeStart weekFields
+
+-- | Gets all the start times for each day.
+checkTimeStart :: [[Double]] -> [String]
+checkTimeStart day = map getStr ([head sortedList] ++ getStartConsecutives sortedList)
+    where
+    sortedList = sort $ map (!! 1) day
+
+-- | Gets the start times that are not the very first start time. 
+getStartConsecutives :: [Double] -> [Double]
+getStartConsecutives lst = filter (/= 30.0) ([if lst !! i == (lst !! (i + 1)) - 0.5 then 30.0 else (lst !! (i + 1))| i <- [0 .. l]])
+    where
+    l = (length lst) - 2
+
+-- ** End time
+
+-- | Obtains a list of end times for each course.
+endTime :: [[[Double]]] -> [[String]]
+endTime weekFields = map checkTimeEnd weekFields
+
+-- | Gets all the end times for each day.
+checkTimeEnd :: [[Double]] -> [String]
+checkTimeEnd day = map getStr (map (+ 0.5) (getEndConsecutives sortedList ++ [last sortedList]))
+    where
+    sortedList = sort $ map (!! 1) day
+
+-- | Gets the end times that are not the very first end time. 
+getEndConsecutives :: [Double] -> [Double]
+getEndConsecutives lst = filter (/= 30.0) ([if lst !! i == (lst !! (i + 1)) - 0.5 then 30.0 else lst !! i| i <- [0 .. l]])
+    where
+    l = (length lst) - 2
+
+-- ** Function that works for both start and end times
+
+-- | Creates the string time in the following way HourMinutesSeccondsZ.
+-- For instance, 133500Z corresponds to 1:35 pm.
+getStr :: Double -> String
+getStr fullTime = if minutes == 0
+                  then hour ++ "0000"
+                  else hour ++ ratio minutes ++ "00"
+    where
+    hours = splitOn "." (show fullTime)
+    hour = hours !! 0
+    minutes = read (hours !! 1) :: Int
+
+-- | Gets the minutes out of a decimal part of my time.
+ratio :: Int -> String
+ratio decimal = if minutes >= 10 then show minutes else "0" ++ show minutes
+    where
+    minutes = decimal * 6
+
+-- ** Start/End date
+
+-- | Obtains all the dates for each course depending on its session.
+getDates :: String -> [[[Double]]] -> [[(String, String)]]
+getDates session dataInOrder = map (checkSession session) dataInOrder
+    where
+    checkSession session1 courseFields = if session1 == "Y" 
+                                        then [halfFall courseFields, halfWinter courseFields]
+                                        else [full session1 courseFields]
+    halfFall courseFields = getDate "F" courseFields
+    halfWinter courseFields = getDate "S" courseFields
+    full session1 courseFields = getDate session1 courseFields
+
+-- | Gives the appropiate starting date for the course.
+-- | Obtains the starting date for each day.
+getDate :: String -> [[Double]] -> (String, String)
```

Comment:
Some of these functions can be given better names. Don't be afraid to give longer names, such that the longer name provides a reader with a quicker/deeper understanding of what the code does. In cases like this, it might also help to add information about the arguments (Some of the files in the Svg module are good examples). For instance, the documentation states something about a course, but I don't see how a course would be represented in the arguments. (This could also be a reflection on how Courseography should make better use of type synonyms.)